File: ordinalSuffix.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

void ordinalSuffix(char *number);

int main(void)
{

    char number[50];
    printf("Enter your number: ");
    scanf("%s", number);

    ordinalSuffix(number);

    return 0;
}

void ordinalSuffix(char *number)
{
    int len = strlen(number);

    int lastTwoDigits = atoi(number + (len - 2));
    int lastOneDigit = atoi(number + (len - 1));
    
    if (lastTwoDigits == 11 || lastTwoDigits == 12 || lastTwoDigits == 13)
    {
        printf("%sth\n", number);
    }
    else
    {
        switch (lastOneDigit)
        {
            case 1: printf("%sst\n", number); break;
            case 2: printf("%snd\n", number); break;
            case 3: printf("%srd\n", number); break;
            default: printf("%sth\n", number); break;
        }
    }
}

File: ordinalSuffix.py
def ordinalSuffix(number):
    last_two_digits = number % 100
    last_one_digit = number % 10

    if last_two_digits in [11, 12, 13]:
        return f"{number}th"
    
    match last_one_digit:
        case 1: return f"{number}st"
        case 2: return f"{number}nd"
        case 3: return f"{number}rd"
        case _: return f"{number}th"

number = int(input("Enter your number: "))
print(ordinalSuffix(number))